### PR TITLE
JAMES-4030 Carry over Bouncer exceptions

### PR DESCRIPTION
**Pull request by chibenwa**
This results on a nack of the underlying mailqueue, thus not loosing the email

### CODE REVIEW

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java [hunk 1]
```diff
@@ -126,11 +126,13 @@ void dsnBounceMailetBouncedMailShouldBeAdressedToTheSenderInEnvelopeAndHeader(@T
...
             );
 
-        MimeMessage mimeMessage = MimeMessageUtil.mimeMessageFromString(testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+        String string = testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
             .login(SENDER, SENDER_PASSWORD)
             .select(TestIMAPClient.INBOX)
             .awaitMessage(awaitAtMostOneMinute)
-            .readFirstMessage());
+            .readFirstMessage();
+        System.out.println(string);
```
Comment (Arsnael) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java:134]: debug spotted

File: server/mailet/mailetcontainer-impl/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailetContext.java [hunk 2]
```diff
@@ -377,7 +377,9 @@ public void sendMail(MailAddress sender, Collection<MailAddress> recipients, Mim
 
     @Override
     public void sendMail(Mail mail) throws MessagingException {
-        sendMail(mail, Optional.ofNullable(mail.getState()).orElse(Mail.DEFAULT));
+        String state = Optional.ofNullable(mail.getState()).orElse(Mail.DEFAULT);
+        System.out.println("Send amil " + state);
```
Comment (Arsnael) [inline server/mailet/mailetcontainer-impl/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailetContext.java:381]: debug spotted

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java [hunk 3]
```diff
@@ -295,6 +295,7 @@ public Optional<String> getSubjectPrefix(Mail newMail, String subjectPrefix, Mai
 
     @Override
     public void service(Mail originalMail) throws MessagingException {
+        System.out.println(originalMail.getState());
```
Comment (Arsnael) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java:298]: debug spotted

**Review by Arsnael — COMMENTED** [review]
And checkstyle issues

Is this still a WIP with all those debugs on the last commit, or those have been forgotten?

**Comment by chibenwa** [comment]
forgotten

**Review by quantranhong1999 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**